pkg/controller/utils/capability.go

```diff
@@ -74,13 +86,19 @@ func NewCapabilityComponentDef(componentDefinition *v1beta1.ComponentDefinition)
 		def.WorkloadType = util.ReferWorkload
 		def.WorkloadDefName = componentDefinition.Spec.Workload.Type
 	}
-	if componentDefinition.Spec.Schematic != nil && componentDefinition.Spec.Schematic.HELM != nil {
-		def.WorkloadType = util.HELMDef
-		def.Helm = componentDefinition.Spec.Schematic.HELM
-	}
-	if componentDefinition.Spec.Schematic != nil && componentDefinition.Spec.Schematic.KUBE != nil {
-		def.WorkloadType = util.KubeDef
-		def.Kube = componentDefinition.Spec.Schematic.KUBE
+	if componentDefinition.Spec.Schematic != nil {
+		if componentDefinition.Spec.Schematic.HELM != nil {
+			def.WorkloadType = util.HELMDef
+			def.Helm = componentDefinition.Spec.Schematic.HELM
+		}
+		if componentDefinition.Spec.Schematic.KUBE != nil {
+			def.WorkloadType = util.KubeDef
+			def.Kube = componentDefinition.Spec.Schematic.KUBE
+		}
+		if componentDefinition.Spec.Schematic.Terraform != nil {
+			def.WorkloadType = util.TerraformDef
```

Comment:
In fact, L99-L100 is covered.
![image](https://user-images.githubusercontent.com/2805315/120291885-5e9ca680-c2f6-11eb-8ae1-f80fe774fbc3.png)